### PR TITLE
Added per camera filters demo.

### PR DESCRIPTION
**Pull request by mrcdk**
It's the same demo as PostProcess :sweat_smile: 

It needs this PR first https://github.com/HaxeFlixel/flixel/pull/1635

### CODE REVIEW

**Comment by Gama11** [comment]
Does this demo compile for you? Doesn't for me and doesn't for Travis - OpenFL doesn't like `<assets>` nodes pointing to non-existant folders:

`Error: Could not find asset path "assets"`

File: Effects/Filters/source/PlayState.hx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+				filters.remove(filter);
+			}
+		}
+	}
+	
+	override public function update(elapsed:Float):Void 
+	{
+		super.update(elapsed);
+		
+		for (filter in filterMap.iterator()) {
```
Comment (Gama11) [inline Effects/Filters/source/PlayState.hx]: Why the explicit `iterator` call?

**Comment by Gama11** [comment]
I don't like the redundancy of having two demos that are almost exactly the same. Is there still a reason to keep the PostProcess API around?

And more importantly, this doesn't really showcase the advantage of this approach (per-camera filters) very well.

File: Effects/Filters/source/shaders/Grain.hx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import openfl.display.Shader;
+
+/**
+ * ...
+ * @author MrCdK
+ */
+class Grain extends Shader
+{
+	
+	@fragment var code = '
```
Comment (Gama11) [inline Effects/Filters/source/shaders/Grain.hx]: No way to load this at runtime via `openfl.Assets.getText()`?

Comment (mrcdk) [inline Effects/Filters/source/shaders/Grain.hx]: No, shaders are a compile time feature, I haven't choose how to load them from a file yet.

**Comment by mrcdk** [comment]
I don't have the time to work on another demo from scratch, this demo shows how to apply the filters to a camera or to the full game which is the main difference between `PostProcess` and this.

File: Effects/Filters/source/shaders/Grain.hx (hunk 2 above)
Comment (Gama11) [inline Effects/Filters/source/shaders/Grain.hx]: Build macros?

Comment (mrcdk) [inline Effects/Filters/source/shaders/Grain.hx]: I know, I haven't chosen how to implement it on OpenFL.

**Comment by Gama11** [comment]
Just noticed some strange behavior in the demo regarding focus loss (neko and cpp, openfl 3.3.9, lime 2.6.9):

1. Click outside of the demo
2. Click the demo window to focus it again - nothing happens
3. Click outside and inside the window a second time, now it resumes, but the cursor is missing

**Comment by mrcdk** [comment]
Isn't that a known issue with openfl next? I've always had that problem.

**Comment by Gama11** [comment]
Dunno, haven't tested next much lately. If it's known, is there an open issue on the repo?

**Pull request merged**